www/common/outer/async-store.js

```diff
@@ -1287,6 +1300,8 @@ define([
                     });
                     return;
                 }
+            } else if (inTrash) {
+                store.manager.user.userObject.restore(path[0]);
```

Comment:
If I understand correctly (I didn't test), it seems this is going to restore any pads from your drive that you open. This function is called everytime someone opens a document, updates its title or tries to store it. This isn't something we really want (trashed document should stay there unless there is an explicit action to restore them).